storage-provider/server/src/pipeline/mod.rs

```diff
@@ -667,36 +667,33 @@ async fn submit_windowed_post(
         return Err(PipelineError::DeadlineStateNotFound);
     };
 
-    if deadline_state.partitions.len() > 1 {
-        todo!("I don't know what to do: polka-storage#595");
-    }
     if deadline_state.partitions.len() == 0 {
         tracing::info!("There are not partitions in this deadline yet. Nothing to prove here.");
         schedule_post(state, deadline_index)?;
         return Ok(());
     }
 
     let partitions = deadline_state.partitions.keys().cloned().collect();
-    let (_partition_number, PartitionState { sectors }) = deadline_state
-        .partitions
-        .first_key_value()
-        .expect("1 partition to be there");
+    let mut all_sectors = BTreeSet::new();
+    for (_, PartitionState { sectors }) in deadline_state.partitions {
+        all_sectors.extend(sectors);
+    }
```

Comment:
How about something like
```suggestion
    let all_sectors = BTreeSet::from_iter(deadline_state.partitions.iter().map(|(_, PartitionState { sectors })| sectors));
```

Reply:
Close enough! `flat_map` and it works.